pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -458,7 +458,8 @@ protected CompletableFuture<Void> doAcknowledge(MessageId messageId, AckType ack
             Consumer individualConsumer = consumers.get(topicMessageId.getTopicPartitionName());
             if (individualConsumer != null) {
                 MessageId innerId = topicMessageId.getInnerMessageId();
-                return individualConsumer.acknowledgeCumulativeAsync(innerId);
+                return individualConsumer.acknowledgeCumulativeAsync(innerId)
+                        .thenAccept(__ -> unAckedMessageTracker.removeMessagesTill(topicMessageId));
```

Comment:
The `unAckedMessageTracker` will have the message IDs from all partitions. If we use `removeMessagesTill` here, the message acknowledgment for partition-0 will also remove the message IDs of other partitions, right?